filer/fields/file.py

```diff
@@ -41,8 +47,9 @@ def render(self, name, value, attrs=None):
         if not related_url:
             related_url = reverse('admin:filer-directory_listing-last')
         params = self.url_parameters()
+        params['_pick'] = 'file'
         if params:
-            lookup_url = '?' + '&amp;'.join(['%s=%s' % (k, v) for k, v in list(params.items())])
+            lookup_url = '?' + urlencode(sorted(params.items()))
```

Comment:
urlencode does not do any encoding for you, if there's a non-ascii character in any of these then 500 error.